appinventor/blocklyeditor/src/blocks/helpers.js

```diff
@@ -96,5 +97,28 @@ Blockly.Blocks['helpers_dropdown'] = {
       }
     }
     return options;
+  },
+
+  typeblock: function() {
+    var db = Blockly.mainWorkspace.getComponentDatabase();
+    var tb = [];
+
+    db.forEachOptionList(function(optionList) {
+      for (var i = 0, option; option = optionList.options[i]; i++) {
+        var tag = db.getInternationalizedOptionListTag(optionList.tag);
+        var key = optionList.tag + option.name;
+        var i18nName = db.getInternationalizedOptionName(key, option.name);
+        tb.push({
+          // TODO: This doesn't handle rtl langs, anyway to fix that?
+          translatedName: tag + ": " + i18nName,
```

Comment:
In RTL languages the whole page text will be reversed (eventually), so the character order here should still render properly.

Reply:
Although I do know there's something peculiar around punctuation. Given that we don't currently support RTL languages I would say this is fine for now.

Reply:
Do we need the colon here? It isn't present in the block. That could confuse people.

Reply:
I decided to remove it :D